vault/plugin_catalog_test.go

```diff
@@ -180,3 +183,98 @@ func TestPluginCatalog_List(t *testing.T) {
+	}
+
+	// register plugins
+	TestAddTestPlugin(t, core, "mux-postgres", consts.PluginTypeUnknown, "TestPluginCatalog_PluginMain_PostgresMultiplexed", []string{}, "")
+	TestAddTestPlugin(t, core, "single-postgres-1", consts.PluginTypeUnknown, "TestPluginCatalog_PluginMain_Postgres", []string{}, "")
+	TestAddTestPlugin(t, core, "single-postgres-2", consts.PluginTypeUnknown, "TestPluginCatalog_PluginMain_Postgres", []string{}, "")
+
+	// run plugins
+	if _, err := core.pluginCatalog.NewPluginClient(context.Background(), testPluginClientConfig(t, "mux-postgres")); err != nil {
+		t.Fatal(err)
+	}
+	if _, err := core.pluginCatalog.NewPluginClient(context.Background(), testPluginClientConfig(t, "mux-postgres")); err != nil {
+		t.Fatal(err)
+	}
+	if _, err := core.pluginCatalog.NewPluginClient(context.Background(), testPluginClientConfig(t, "single-postgres-1")); err != nil {
+		t.Fatal(err)
+	}
+	if _, err := core.pluginCatalog.NewPluginClient(context.Background(), testPluginClientConfig(t, "single-postgres-2")); err != nil {
+		t.Fatal(err)
+	}
+
+	externalPlugins := core.pluginCatalog.externalPlugins
+	if len(externalPlugins) != 3 {
+		t.Fatalf("expected externalPlugins map to be of len 3 but got %d", len(externalPlugins))
+	}
+
+	// check connections map
+	if len(externalPlugins["mux-postgres"].connections) != 2 {
+		t.Fatalf("expected multiplexed external plugin's connections map to be of len 2 but got %d", len(externalPlugins["mux-postgres"].connections))
+	}
+	if len(externalPlugins["single-postgres-1"].connections) != 1 {
+		t.Fatalf("expected multiplexed external plugin's connections map to be of len 2 but got %d", len(externalPlugins["mux-postgres"].connections))
+	}
+	if len(externalPlugins["single-postgres-2"].connections) != 1 {
+		t.Fatalf("expected multiplexed external plugin's connections map to be of len 2 but got %d", len(externalPlugins["mux-postgres"].connections))
+	}
+
+	// check multiplexing support
+	if !externalPlugins["mux-postgres"].multiplexingSupport {
+		t.Fatalf("expected external plugin to be multiplexed")
+	}
+	if externalPlugins["single-postgres-1"].multiplexingSupport {
+		t.Fatalf("expected external plugin to be non-multiplexed")
+	}
+	if externalPlugins["single-postgres-2"].multiplexingSupport {
+		t.Fatalf("expected external plugin to be non-multiplexed")
+	}
+}
+
+func TestPluginCatalog_PluginMain_Postgres(t *testing.T) {
+	if os.Getenv(pluginutil.PluginVaultVersionEnv) == "" {
+		return
+	}
+
+	dbType, err := postgresql.New()
+	if err != nil {
+		t.Fatalf("Failed to initialize postgres: %s", err)
+	}
+
+	v5.Serve(dbType.(v5.Database))
+}
+
+func TestPluginCatalog_PluginMain_PostgresMultiplexed(t *testing.T) {
+	if os.Getenv(pluginutil.PluginVaultVersionEnv) == "" {
+		return
+	}
+
+	v5.ServeMultiplex(postgresql.New)
+}
+
+func testPluginClientConfig(t *testing.T, pluginName string) pluginutil.PluginClientConfig {
```

Comment:
Since there's no failure case here or test-specific checks, would it be fine to drop `testing.T` from this func?